### PR TITLE
fix: Fallback calculation returns correct value

### PR DESCRIPTION
**Pull request by JorTurFer**
_Provide a description of what has been changed_

### Checklist

- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/4249

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 ## History
 
 - [Unreleased](#unreleased)
+- [v2.9.3](#v293)
```
Comment (zroubalik) [inline CHANGELOG.md:19]: could you please exclude this from this PR?
it is unrelated, I will open a new PR to fix this.

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zroubalik) [reply CHANGELOG.md:19]: https://github.com/kedacore/keda/pull/4264

Comment (JorTurFer) [reply CHANGELOG.md:19]: sure, omw

Comment (JorTurFer) [reply CHANGELOG.md]: done :)

**Review by zroubalik — APPROVED** [review]

**Pull request merged**